### PR TITLE
Fhirservice improvements

### PR DESCRIPTION
**Pull request by ThishaniLucas**

### CODE REVIEW

File: fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        public function createInterceptors() returns [r4:FHIRResponseErrorInterceptor, r4:FHIRResponseInterceptor] {
+            return [new r4:FHIRResponseErrorInterceptor(), new r4:FHIRResponseInterceptor(apiConfig)];
+        }
+
+        isolated resource function get [string... paths](http:Request req, http:RequestContext ctx) returns any|error {
+
+            Service fhirService = self.holder.getFhirServiceFromHolder();
+            string path = req.extraPathInfo;
+            string[] split = getSplitPath(path);
+            handle? resourceMethod = getResourceMethod(fhirService, split, "get");
```
Comment (nirmal070125) [inline fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal:36]: Can you extract them to a function?

File: fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                        return r4:handleErrorResponse(executeResourceResult);
+                    }
+
+                }
+                return executeResourceResult;
+            } else {
+                return r4:createFHIRError(string `Path not found: ${path}`, r4:CODE_SEVERITY_ERROR, r4:TRANSIENT, httpStatusCode = http:STATUS_NOT_FOUND);
+            }
+        }
+        isolated resource function post [string... paths](http:Request req, http:RequestContext ctx) returns anydata|error {
```
Comment (sameeragunarathne) [inline fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal:73]:
```suggestion
        
        isolated resource function post [string... paths](http:Request req, http:RequestContext ctx) returns anydata|error {
```

File: fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/regex;
+import ballerinax/health.fhir.r4;
+
+isolated function getHttpService(Holder h, r4:ResourceAPIConfig apiConfig) returns http:Service {
```
Comment (sameeragunarathne) [inline fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal:21]: Let's add bal docs

File: fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+            } else {
+                return r4:createFHIRError(string `Path not found: ${path}`, r4:CODE_SEVERITY_ERROR, r4:TRANSIENT, httpStatusCode = http:STATUS_NOT_FOUND);
+            }
+
+        }
+    };
+    return httpService;
+}
+
+isolated function getSplitPath(string restPath) returns string[] {
```
Comment (sameeragunarathne) [inline fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal:108]: Let's add bal docs

File: fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal (hunk 3 above)
Comment (ThishaniLucas) [reply fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal:22]: added comment.

File: fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal (hunk 2 above)
Comment (ThishaniLucas) [reply fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal:75]: done

File: fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal (hunk 4 above)
Comment (ThishaniLucas) [reply fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal:111]: added a comment

**Review by nirmal070125 — APPROVED** [review]

**Pull request merged**